source/common/common/utility.h

```diff
@@ -551,7 +552,40 @@ template <class Value> struct TrieEntry {
  * A trie used for faster lookup with lookup time at most equal to the size of the key.
  */
 template <class Value> struct TrieLookupTable {
-
+  /**
+   * A custom destructor helps avoid a stack overflow issue when trie depth is great and default
+   * destructor recursively collects unique_ptr.
+   */
+  ~TrieLookupTable() {
+    // Use a stack to store non-nullptr nodes which should be deleted, and their children will be
+    // pushed into this stack and will be deleted later.
+    std::stack<TrieEntry<Value>*> to_delete;
+    // Release all pointers in root_ and push them into the stack.
+    for (unsigned long i = 0; i < root_.entries_.size(); i++) {
+      auto entry = std::move(root_.entries_[i]); // This entry in trie will be nullptr now.
+      if (entry != nullptr) {
+        to_delete.push(
+            entry.release()); // Raw points will be inside the stack and will be deleted later.
```

Comment:
nit: single line 
(Also the comment may be redundant since there's a nice comment about what `to_delete` holds.